pkg/provider/kubernetes/credentials.go

```diff
@@ -411,18 +411,46 @@ func createOrUpdateKubevirtSecret(ctx context.Context, seedClient ctrlruntimecli
 
 func createVSphereSecret(ctx context.Context, seedClient ctrlruntimeclient.Client, cluster *kubermaticv1.Cluster) (bool, error) {
 	spec := cluster.Spec.Cloud.VSphere
+	// for existing clusters we need to fetch existing credential data to set infra user and password if empty
+	// this is caused by a bug fix related to credential separation
+	// see https://github.com/kubermatic/kubermatic/issues/13534 and https://github.com/kubermatic/kubermatic/issues/13995
+	// for further information
+	secretKeySelector := provider.SecretKeySelectorValueFuncFactory(ctx, seedClient)
 
+	if val, _ := secretKeySelector(spec.CredentialsReference, resources.VsphereInfraManagementUserUsername); val == "" {
+		if val, _ := secretKeySelector(spec.CredentialsReference, resources.VsphereUsername); val != "" {
+			spec.Username = val
+		}
+	}
+
+	if val, _ := secretKeySelector(spec.CredentialsReference, resources.VsphereInfraManagementUserPassword); val == "" {
+		if val, _ := secretKeySelector(spec.CredentialsReference, resources.VspherePassword); val != "" {
+			spec.Password = val
+		}
+	}
 	// already migrated
 	if spec.Username == "" && spec.Password == "" && spec.InfraManagementUser.Username == "" && spec.InfraManagementUser.Password == "" {
 		return false, nil
 	}
 
+	// if credentials for infrastructure management are set explicit we need to overwrite username and password with these
+	// to mount them properly to machine-controller and osm
+	vsphereInfraUser := spec.Username
+	if spec.InfraManagementUser.Username != "" {
+		vsphereInfraUser = spec.InfraManagementUser.Username
+	}
+
+	vsphereInfraPassword := spec.Password
+	if spec.InfraManagementUser.Password != "" {
+		vsphereInfraPassword = spec.InfraManagementUser.Password
+	}
+
```

Comment:
vsphereInfraPassword and vsphereInfraPassword should be empty if they are not specified. By your logic, they will always be configured as `VsphereInfraManagementUserUsername=username` and `VsphereInfraManagementUserPassword=password` when not specified, which is wrong, unfortunately.

This can be highly confusing for the end users since they never specified the infra management credentials but somehow they have been injected by KKP.